File: bfs.py

from collections import deque

def bfs(graph, start):
    visited = set()
    queue = deque([start])
    visited.add(start)
    while queue:
        vertex = queue.popleft()
        print(vertex)
        for neighbor in graph.get(vertex, []):
            if neighbor not in visited:
                visited.add(neighbor)
                queue.append(neighbor)

# Example Usage:
graph_bfs = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': ['F'],
    'D': [],
    'E': ['F'],
    'F': []
}
print("BFS Traversal:")
bfs(graph_bfs, 'A')

File: bfs.js

function bfs(graph, start) {
    const visited = new Set();
    const queue = [start];
    visited.add(start);

    let traversal = [];

    while (queue.length > 0) {
        const vertex = queue.shift();
        traversal.push(vertex);

        const neighbors = graph[vertex] || [];
        for (const neighbor of neighbors) {
            if (!visited.has(neighbor)) {
                visited.add(neighbor);
                queue.push(neighbor);
            }
        }
    }
    return traversal;
}

// Example Usage:
const graph_bfs = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': ['F'],
    'D': [],
    'E': ['F'],
    'F': []
};
console.log("BFS Traversal:");
console.log(bfs(graph_bfs, 'A'));
